### PR TITLE
Update README.md and Add Module.md and Package.md files

### LINKED ISSUE #3176: Add documentation for serdes module
**Description:**
$Subject

**Describe your task(s)**

- [x] Update README.md
- [x] Add Module.md
- [x] Add Package.md

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose
Fixes: [#3176](https://github.com/ballerina-platform/ballerina-standard-library/issues/3176)

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,46 @@ Ballerina SerDes Library
...
+
+```ballerina
+// Define a type which is a subtype of anydata.
+type Student record {
+    int id;
+    string name;
+    decimal fees;
+};
+
+// Create a serialization object by passing the typedesc.
```
Comment (ThisaruGuruge) [inline README.md]:
```suggestion
// Create a schema object by passing the type.
```

File: README.md [hunk 2]
```diff
@@ -9,6 +9,46 @@ Ballerina SerDes Library
...
+
+```ballerina
+Student studentValue = {
+    id: 7894,
+    name: "Liam",
+    fees: 24999.99
+};
+
+// Serialize the record value to bytes.
+byte[] serializedValue = check schema.serialize(studentValue);
```
Comment (ThisaruGuruge) [inline README.md]: We usually don't define variable names like `studentValue`, right?

(You can come up with a better name than `bytes`.)

```suggestion
Student student = {
    id: 7894,
    name: "Liam",
    fees: 24999.99
};

// Serialize the record value to bytes.
byte[] bytes = check schema.serialize(student);
```

File: README.md [hunk 3]
```diff
@@ -9,6 +9,46 @@ Ballerina SerDes Library
...
+```
+A value having the same type as the provided typedesc can be serialized by invoking the `serialize` method on the previously instantiated `serdes:Proto3Schema` object. The underlying implementation uses the previously generated proto3 schema to serialize the provided value.
```
Comment (ThisaruGuruge) [inline README.md]: Let's use `type` here.

```suggestion
A value having the same type as the provided type can be serialized by invoking the `serialize` method on the previously instantiated `serdes:Proto3Schema` object. The underlying implementation uses the previously generated proto3 schema to serialize the provided value.
```

File: README.md [hunk 4]
```diff
@@ -9,6 +9,46 @@ Ballerina SerDes Library
...
+
+#### Serialization
```
Comment (ThisaruGuruge) [inline README.md]:
```suggestion
#### Deserialization
```

File: README.md [hunk 5]
```diff
@@ -9,6 +9,46 @@ Ballerina SerDes Library
...
+
+```ballerina
+Student deserializedValue = check schema.deserialize(serializedValue);
+```
```
Comment (ThisaruGuruge) [inline README.md]: Let's provide a complete example for this, by creating the schema and deserializing it, considering this is a separate section.

File: README.md (hunk 1 above)
Comment (MohamedSabthar) [reply README.md]: Updated.

File: README.md (hunk 3 above)
Comment (MohamedSabthar) [reply README.md]: Updated

File: README.md (hunk 4 above)
Comment (MohamedSabthar) [reply README.md]: Updated

File: README.md (hunk 2 above)
Comment (MohamedSabthar) [reply README.md]: Updated, sticking to `bytes` for now

File: README.md (hunk 5 above)
Comment (MohamedSabthar) [reply README.md]: Shall we do it like this ?

```ballerina
type Student record {
    int id;
    string name;
    decimal fees;
};

byte[] bytes = readSerializedDataToByteArray();
serdes:Proto3Schema schema = check new (Student);
Student student = check schema.deserialize(bytes);
```

Comment (MohamedSabthar) [reply README.md]: Updated as above

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**